### PR TITLE
Possible type confusions

### PR DESCRIPTION
**Pull request by vincentgoat**
Signed-off-by: vincentgoat <linguohui1@huawei.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Several possible type confusions were discovered in the KubeEdge code base. These should be avoided by checking for an error when type casting.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/pkg/eventbus/eventbus.go [hunk 1]
```diff
@@ -124,12 +124,24 @@ func (eb *eventbus) pubCloudMsgToEdge() {
...
+				continue
+			}
+			topic, ok := message["topic"].(string)
+			if !ok {
+				klog.Errorf("Message topic body type is %T and not string type", message["topic"])
+				continue
+			}
+			payload, err := json.Marshal(&message)
+			if err != nil {
+				klog.Errorf("marshal message %v error: %v", topic, err)
```
Comment (gy95) [inline edge/pkg/eventbus/eventbus.go:142]: print topic or message?

File: cloud/pkg/router/listener/http.go [hunk 2]
```diff
@@ -78,7 +78,11 @@ func (rh *RestHandler) RemoveListener(key interface{}) {
 func (rh *RestHandler) matchedPath(uri string) (string, bool) {
 	var candidateRes string
 	rh.handlers.Range(func(key, value interface{}) bool {
-		pathReg := key.(string)
+		pathReg, ok := key.(string)
+		if !ok {
+			klog.Errorf("key type %T error", key)
+			return true
```
Comment (gy95) [inline cloud/pkg/router/listener/http.go:84]: return false

**Review by gy95 — APPROVED** [review]

File: edge/pkg/eventbus/eventbus.go (hunk 1 above)
Comment (vincentgoat) [reply edge/pkg/eventbus/eventbus.go:142]: > print topic or message?

The topic is fine, avoid printing any sensitive message.

Comment (gy95) [reply edge/pkg/eventbus/eventbus.go:142]: ok

File: cloud/pkg/router/listener/http.go (hunk 2 above)
Comment (vincentgoat) [reply cloud/pkg/router/listener/http.go:84]: The return value should be `true`, according to the Range function.
image

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**